src/server.c

```diff
@@ -330,6 +330,8 @@ static int on_vencrypt_plain_auth_message(struct nvnc_client* client)
 	username[MIN(ulen, sizeof(username) - 1)] = '\0';
 	password[MIN(plen, sizeof(password) - 1)] = '\0';
 
+	strncpy(client->username, username, sizeof(client->username));
```

Comment:
I haven't added `strlcpy` to this project, so we manually zero terminate after `strncpy`.